### ISSUE TITLE
Unable to locate sign at -58, 4, -325

### ISSUE THREAD

**Issue by rom1504** [PrismarineJS/node-minecraft-protocol#167]
Connecting with the vanilla client through the proxy gives these error messages in the chat.

tested on a spigot server and on us.mineplex.com

**Comment by rom1504** [comment]
> +SpaceManiac the client prints that if it receives a sign text update before it knows there's a sign there

**Comment by roblabla** [comment]
actually, now that you mention it.... it _is_ possible to send packets out-of-order.

**Comment by roblabla** [comment]
the huge packets (e.g. the map) are decompressed then re-compressed : https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/src/protocol.js#L230, which are asynchronous operations. Assuming those happen outside the event loop, it is entirely possible for other packets to be processed before this one.

**Comment by rom1504** [comment]
```js
client<-server: play.0x33 : {"id":51,"state":"play","location":{"x":23,"y":54,"z":-18},"text1":"\"\"","text2":"\"\"","text3":"\"\"","text4":"\"\""}
client<-server: play.0x33 : {"id":51,"state":"play","location":{"x":23,"y":54,"z":-19},"text1":"\"\"","text2":"\"\"","text3":"\"\"","text4":"\"\""}
// ...
client<-server: play.0x26 : Packet too big
```
> roblabla> the world packets are sent after the sign packets
> roblabla> uncompressing is probably the culprit

**Comment by rom1504** [comment]
Somehow this doesn't happen with a vanilla server.

**Comment by rom1504** [comment]
Oh fixing the bug said in #168 fix that issue with a spigot server. With us.mineplex.com I got a new error when connecting : "Internal Exception: io.netty.handler.codec.DecoderException: java.io.IOException: Bad packet id 120".

**Comment by rom1504** [comment]
So I think we need : 
* a emit queue (for this https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/src/client.js#L98)
* a write queue (for this https://github.com/PrismarineJS/node-minecraft-protocol/blob/master/examples/proxy/proxy.js#L107) in the proxy

Not sure exactly what kind of queue that would be

**Comment by roblabla** [comment]
Fixed with #174
